openmp/libomptarget/plugins-nextgen/common/src/JIT.cpp

```diff
@@ -330,24 +330,18 @@ JITEngine::process(const __tgt_device_image &Image,
   return &Image;
 }
 
-bool JITEngine::checkBitcodeImage(const __tgt_device_image &Image) {
+Expected<bool> JITEngine::checkBitcodeImage(StringRef Buffer) const {
   TimeTraceScope TimeScope("Check bitcode image");
 
-  if (!isImageBitcode(Image))
-    return false;
-
-  StringRef Data(reinterpret_cast<const char *>(Image.ImageStart),
-                 target::getPtrDiff(Image.ImageEnd, Image.ImageStart));
-  auto MB = MemoryBuffer::getMemBuffer(Data, /*BufferName=*/"",
-                                       /*RequiresNullTerminator=*/false);
-  if (!MB)
-    return false;
+  assert(identify_magic(Buffer) == file_magic::bitcode &&
+         "Input is not bitcode");
 
   LLVMContext Context;
-  SMDiagnostic Diagnostic;
-  std::unique_ptr<Module> M =
-      llvm::getLazyIRModule(std::move(MB), Diagnostic, Context,
-                            /*ShouldLazyLoadMetadata=*/true);
+  auto ModuleOrErr = getLazyBitcodeModule(MemoryBufferRef(Buffer, ""), Context,
+                                          /*ShouldLazyLoadMetadata=*/false);
```

Comment:
Can the construction of the `MemoryBufferRef` fail?

I probably don't know enough about the different APIs. Can you tell me why the last parameter here was changed to `false`?

Reply:
No, a memory buffer reference is just a reference to existing memory. The second parameter is the buffer's identifier. E.g. if it was opened from a file it would be the filename.

Reply:
Thanks for explaining, and sorry for confusion. I was asking for the parameter about "ShouldLazyLoadMetadata". Why has that changed to false?

Reply:
Oh, good catch, that was unintentional. We're only interested in the Triple so we should skip loading the entire table.